test/e2e/framework/service/jig.go

```diff
@@ -82,6 +106,68 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// allocatePort reserves the port provided as input.
+// If an invalid port was provided or if the port is already reserved, it returns false
+func (s *staticPortRange) allocatePort(port int) bool {
+	s.Lock()
+	defer s.Unlock()
+	port -= s.baseport
+	if port < 0 || port > s.length || s.reservedPorts.Has(port) {
+		return false
+	}
+	s.reservedPorts.Insert(port)
+	return true
+}
+
+// nextFreePort allocates a free port from the range and returns its number and true
+// if none is available then it returns -1 and false
+func (s *staticPortRange) nextFreePort() (int, bool) {
+	s.Lock()
+	defer s.Unlock()
+	// start in a random offset
+	start := rand.Intn(s.length)
+	for i := 0; i < s.length; i++ {
+		port := (start + i) % s.length
+		if port < s.length && port >= 0 && !s.reservedPorts.Has(port) {
```

Comment:
```suggestion
		!s.reservedPorts.Has(port) {
```

you are calculating the modulo and the value is always > 0,